### PR TITLE
[DEPRECATE] SimpleFS in favor of NIOFS

### LINKED ISSUE #1072: Deprecate SimpleFS File System Format
Lucene 9.0 removes the SimpleFS File System Format. This needs to be deprecated in OpenSearch 1.x in favor of NIOFS and officially removed in 2.0.

### PR DESCRIPTION
**Pull request by nknize**
Lucene 9 removes support for SimpleFS File System format. This PR deprecates the SimpleFS format in favor of NIOFS.

resolves #1072

### CODE REVIEW

**Comment by nknize** [comment]
start gradle check

**Comment by nknize** [comment]
start gradle check

**Comment by nknize** [comment]
start gradle check

**Comment by nknize** [comment]
start gradle check

File: server/src/main/java/org/opensearch/cluster/metadata/MetadataUpdateSettingsService.java [hunk 1]
```diff
@@ -252,6 +252,8 @@ public ClusterState execute(ClusterState currentState) {
                     IndexSettings.INDEX_TRANSLOG_RETENTION_SIZE_SETTING.exists(normalizedSettings)) {
                     for (String index : actualIndices) {
                         MetadataCreateIndexService.validateTranslogRetentionSettings(metadataBuilder.get(index).getSettings());
+                        // validate storeType for deprecating index stores
+                        MetadataCreateIndexService.validateStoreTypeSettings(metadataBuilder.get(index).getSettings());
```
Comment (adnapibar) [inline server/src/main/java/org/opensearch/cluster/metadata/MetadataUpdateSettingsService.java:256]: Minimize calls to `metadataBuilder.get(index).getSettings()` ?

**Review by adnapibar — APPROVED** [review]
LGTM 👍  A few minor comments.

What about the usage in the OpenSearch project, are we going to replace those? For instance, 
 
https://github.com/opensearch-project/OpenSearch/blob/4906a3c1b029f55e41bf60ef214ebff61939ba6d/server/src/main/java/org/opensearch/common/settings/KeyStoreWrapper.java#L43

File: plugins/store-smb/src/main/java/org/opensearch/plugin/store/smb/SMBStorePlugin.java [hunk 2]
```diff
@@ -43,11 +46,16 @@
 
 public class SMBStorePlugin extends Plugin implements IndexStorePlugin {
 
+    private static final DeprecationLogger DEPRECATION_LOGGER = DeprecationLogger.getLogger(SMBStorePlugin.class);
+
     @Override
     public Map<String, DirectoryFactory> getDirectoryFactories() {
         final Map<String, DirectoryFactory> indexStoreFactories = new HashMap<>(2);
         indexStoreFactories.put("smb_mmap_fs", new SmbMmapFsDirectoryFactory());
+        DEPRECATION_LOGGER.deprecate(IndexModule.Type.SIMPLEFS.getSettingsKey(), IndexModule.Type.SIMPLEFS.getSettingsKey()
```
[the remark this answers is not in the document]
Comment (nknize) [reply plugins/store-smb/src/main/java/org/opensearch/plugin/store/smb/SMBStorePlugin.java:55]: This message is posted once, when the plugin is loaded and the Factory instance is created. The other message is posted every time the `factory.newFSDirectory` is called (which could, though probably unlikely, happen in a user application) to instantiate a new instance of the SimpleFSDirectory. So I think it's worthy having both.

File: plugins/store-smb/src/main/java/org/opensearch/index/store/smbniofs/SmbNIOFsDirectoryFactory.java [hunk 3]
```diff
@@ -0,0 +1,30 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.index.store.smbniofs;
```
[the remark this answers is not in the document]
Comment (nknize) [reply plugins/store-smb/src/main/java/org/opensearch/index/store/smbniofs/SmbNIOFsDirectoryFactory.java:9]: Yes, I agree. I think maybe we do that refactor in a followup PR? How about in the removal PR?

**Comment by nknize** [comment]
> What about the usage in the OpenSearch project, are we going to replace those? For instance,
> 
> https://github.com/opensearch-project/OpenSearch/blob/4906a3c1b029f55e41bf60ef214ebff61939ba6d/server/src/main/java/org/opensearch/common/settings/KeyStoreWrapper.java#L43

I was going to leave them for full replacement in the Lucene 9 upgrade, but I think it's a better idea to take care of it now. So that's done

**Comment by nknize** [comment]
start gradle check

**Pull request merged**